drivers/Watchdog.h

```diff
@@ -34,10 +44,10 @@ namespace mbed {
  * @code
  *
  * Watchdog watchdog = Watchdog();
- * watchdog.start(2000);
+ * watchdog.register(tid, 2000);
```

Comment:
Why are we doing that? Do we have a use case for thread's support? I don't want force users to care about TID if they don't need to. It's a class why do we need `wd_` prefix? Please don't change the API names if it can be avoided. I know it's on feature branch and we could change the APIs, but best to avoid it if possible.

Reply:
previous review comments on earlier version of watchdog driver requested multithreaded supported driver.
1.Need unique id to differentiate different thread while registering into watchdog driver, so plan to use tid as a key to differentiate multiple threads or if you have any suggestion we can discuss in the design meeting planned today
 2.we cant use start and stop API which is not match functionality because in the new implementation start and stop calls based on registered user/tid so plan to use API name as register,unregister,kick without wd_

Reply:
@rajkan01 - I agree with @bulislaw that users should not be forced to pass thread ID. Using thread ID is great, but it can be done internally in driver using `get_id` of current thread https://github.com/ARMmbed/mbed-os/blob/master/rtos/ThisThread.h#L180.

It will also avoid thread1 - calling `kick` for thread2 and vice versa, as users can do this mistake. Also, if `get_id` is called from ISR context it will return NULL.